### PR TITLE
TESTENG-1565: Add code of conduct and mention coding standards

### PR DESCRIPTION
**Pull request by Sai-Nissankula**
Complete the following tasks before merging in your PR:
- [x] If this is a user facing change, update [version.txt](https://github.com/relativitydev/relativity.testing.framework/blob/master/version.txt) and [CHANGELOG.md](https://github.com/relativitydev/relativity.testing.framework/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by Sai-Nissankula** [comment]
This PR also covers [TESTENG-1566](https://jira.kcura.com/browse/TESTENG-1566) since that was just adding a blurb about that our style guidelines are enforced by style cop.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -92,6 +67,10 @@ While the prerequisites above must be satisfied prior to having your pull reques
 
 #### Styleguides
 
+##### Coding Standards
+
+We have automated as many of our coding standards using StyleCop, but we might identify missing rules, and ask you to fix them in the meantime until we can update StyleCop.
```
Comment (briantoms) [inline CONTRIBUTING.md:72]: Missing `as possible` in the first clause?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,56 +1,31 @@
...
 
 Be sure to also review the CONTRIBUTING.md at the root of the repository you are working in for any information specific to that codebase.
 
+#### Code of Conduct
+
+Before contributing, please review [CODE_OF_CONDUCT.md](https://github.com/relativitydev/relativity.testing.framework/blob/master/CODE_OF_CONDUCT.md).
+
 #### Development
 
 All development work should be done in feature branches off of the master branch.
```
Comment (briantoms) [inline CONTRIBUTING.md:31]: Can external devs create branches, or do they need to work in their own forks?

Comment (paulletang) [reply CONTRIBUTING.md:31]: I tested this out with a couple of PRs.
It's all through forks, they won't have the ability to create branches, unless we want to make that a default permission and/or grant that when asked.

I think that forking is probably the better place to start.
It probably should happen off of a fork of master though, so we should mention something along those lines.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,56 +1,31 @@
...
+:+1::tada: Thanks for taking the time to contribute! :tada::+1:
 
 ## Table Of Contents
 
-- [Contributing to Relativity Testing Framework](#contributing-to-relativity-testing-framework)
-  - [Table Of Contents](#table-of-contents)
-  - [Where to get help](#where-to-get-help)
-  - [What should I know before I get started?](#what-should-i-know-before-i-get-started)
-    - [Relativity Testing Framework Packages](#relativity-testing-framework-packages)
+- [Contributing to Relativity.Testing.Framework](#contributing-to-relativitytestingframework)
```
Comment (paulletang) [inline CONTRIBUTING.md:7]: Why are parts of this moving to the readme?
Am I just not remembering something on the story?
This feels out of scope.

Comment (paulletang) [reply CONTRIBUTING.md:7]: I don't want to discourage the desire to make things better, but it needs to be done in the correct scope for the story.

I would far rather see issues brought up for desired changes than ad hoc out of scope ones.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,59 +1,35 @@
...
-- If you find a **Closed** issue that seems like it is the same thing that you're experiencing, open a new issue and include a link to the original issue in the body of your new one.
-
 ### Code Contributions
 
 Be sure to also review the CONTRIBUTING.md at the root of the repository you are working in for any information specific to that codebase.
 
 #### Development
 
-All development work should be done in feature branches off of the master branch.
+When forking the repository, please make sure you are working of the latest code in the master branch.
```
Comment (Sai-Nissankula) [inline CONTRIBUTING.md:32]: From what I saw on google, "forking" is done at a repo level and not specific branches. I couldnt think of a better to word that use the latest master branch code once forked.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,76 +1,86 @@
...
-
-e.g.
-
-> Adding ObjectTypeGuidAttribute for use in RTF.Api ObjectManager. Adding ObjectTypeGuidResolver and tests.
-
-#### Releasing
-
-Feature branches will create a prerelease package and upload them as an artifact of the build.
-Release branches will publish a golden package to NuGet.
+# Contributing to Relativity.Testing.Framework
```
Comment (Sai-Nissankula) [inline CONTRIBUTING.md:1]: Had weird merge conflict, but now this is resolved. There isnt as many changes as it looks like. The only new stuff are the Code of Conduct + Style cop sections.

File: README.md [hunk 6]
```diff
@@ -4,31 +4,35 @@
...
+  - [Official Support](#official-support)
   - [Documentation](#documentation)
   - [Where to get help](#where-to-get-help)
   - [Build Tasks](#build-tasks)
   - [Local Testing](#local-testing)
     - [Unit Testing](#unit-testing)
+  - [Contributing](#contributing)
   - [Reporting Issues](#reporting-issues)
 
+## Official Support
```
Comment (Sai-Nissankula) [inline README.md:23]: I moved this because its not "Where to get help". This is pretty critical to tell right when someone opens the README to make sure they are using the right versions imo.

File: CODE_OF_CONDUCT.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+# Contributor Covenant Code of Conduct
```
Comment (paulletang) [inline CODE_OF_CONDUCT.md:1]: I think plex had some specific DOs and DONTs they wanted us to include in the contributing guide.

Is there a way to incorporate them in this without making it feel forced?
https://einstein.kcura.com/pages/viewpage.action?spaceKey=PT&title=Best+Practices+for+Relativity+Github+Projects

**Review by paulletang — COMMENTED** [review]
There's two different tasks for the requested changes.
https://jira.kcura.com/browse/TESTENG-1565 and https://jira.kcura.com/browse/TESTENG-1566.

Combining two tasks into one PR increase the blast radius if we need to back something.
It also increases the time and effort required to get the PR reviewed, and refactored with requested changes.

In this specific instance, I think the stylecop/contributing guide changes are good to merge in, but the code of conduct is not, so the entire PR is blocked.

Try to make separate PRs in the future for separate stories/tasks so that we don't run into this going forward.

File: CODE_OF_CONDUCT.md (hunk 7 above)
Comment (Sai-Nissankula) [reply CODE_OF_CONDUCT.md:1]: I found a few that made sense.

**Review by paulletang — APPROVED** [review]
